In [1]:
# Import seaborn and apply its plotting styles
import seaborn as sns
sns.set(font_scale=2, style="white")

import matplotlib
import matplotlib as mpl
import matplotlib.pyplot as plt
import matplotlib.style as style
# set plotting size parameter
plt.rcParams['figure.figsize'] = (17, 7)

# import pandas & numpy library
import pandas as pd
import numpy as np
import scipy.stats as stats

# Statmodels
import statsmodels
import statsmodels.api as sm
import statsmodels.formula.api as smf

## Load the Different Datasets
---

In [2]:
df_FL_Benthic_Cover = pd.read_csv("../00-raw/CRCP_Benthic_Cover_Florida_7018_0ee5_9488.csv")

df_PR_Benthic_Cover = pd.read_csv("../00-raw/CRCP_Benthic_Cover_Puerto_Rico_07c6_3c17_78b6.csv")

df_USVI_Benthic_Cover = pd.read_csv("../00-raw/CRCP_Benthic_Cover_USVI_6400_ac60_8c98.csv")

/tmp/ipykernel_5569/4267141812.py:5: DtypeWarning: Columns (1,2,17,31,32,33,36,37,38) have mixed types. Specify dtype option on import or set low_memory=False.
  df_USVI_Benthic_Cover = pd.read_csv("../00-raw/CRCP_Benthic_Cover_USVI_6400_ac60_8c98.csv")


### Florida Data

In [3]:
df_FL_Benthic_Cover.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13602 entries, 0 to 13601
Data columns (total 35 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   time                 13602 non-null  object 
 1   latitude             13602 non-null  object 
 2   longitude            13602 non-null  object 
 3   REGION               13601 non-null  object 
 4   PRIMARY_SAMPLE_UNIT  13601 non-null  float64
 5   STATION_NR           13601 non-null  float64
 6   YEAR                 13601 non-null  float64
 7   MONTH                13601 non-null  float64
 8   DAY                  13601 non-null  float64
 9   Date_UTC             13602 non-null  object 
 10  HABITAT_CD           13598 non-null  object 
 11  HABITAT_TYPE         13551 non-null  object 
 12  STRAT                13578 non-null  object 
 13  Description          8748 non-null   object 
 14  RUGOSITY_CD          3460 non-null   float64
 15  WTD_RUG              7046 non-null  

In [4]:
df_FL_Benthic_Cover.head()

,time,latitude,longitude,REGION,PRIMARY_SAMPLE_UNIT,STATION_NR,YEAR,MONTH,DAY,Date_UTC,...,DEPTH_STRAT,MIN_DEPTH,MAX_DEPTH,METERS_COMPLETED,COVER_CAT_CD,COVER_CAT_NAME,HARDBOTTOM_P,SOFTBOTTOM_P,RUBBLE_P,accession_url
0,UTC,degrees_north,degrees_east,NaN,NaN,NaN,NaN,NaN,NaN,UTC,...,NaN,m,m,m,NaN,NaN,percent,percent,percent,NaN
1,2016-06-22T00:00:00Z,24.46499,-81.9823,FLK,1702.0,1.0,2016.0,6.0,22.0,2016-06-22T00:00:00Z,...,NaN,7.010399776,7.315199766,15,EUS FAST,Eusmilia fastigiata,1,0,0,https://accession.nodc.noaa.gov/0169401
2,2016-06-22T00:00:00Z,24.46499,-81.9823,FLK,1702.0,1.0,2016.0,6.0,22.0,2016-06-22T00:00:00Z,...,NaN,7.010399776,7.315199766,15,GOR ENCR,Encrusting gorgonian,2,0,0,https://accession.nodc.noaa.gov/0169401
3,2016-06-22T00:00:00Z,24.46499,-81.9823,FLK,1702.0,1.0,2016.0,6.0,22.0,2016-06-22T00:00:00Z,...,NaN,7.010399776,7.315199766,15,GOR GORG,Gorgonians,13,0,0,https://accession.nodc.noaa.gov/0169401
4,2016-06-22T00:00:00Z,24.46499,-81.9823,FLK,1702.0,1.0,2016.0,6.0,22.0,2016-06-22T00:00:00Z,...,NaN,7.010399776,7.315199766,15,MEA MEAN,Meandrina meandrites,1,0,0,https://accession.nodc.noaa.gov/0169401


In [5]:
df_FL_Benthic_Cover.describe()

,PRIMARY_SAMPLE_UNIT,STATION_NR,YEAR,MONTH,DAY,RUGOSITY_CD,WTD_RUG,MAPGRID_NR,SUB_REGION_NR,ZONE_NR,MPA_NR,ADMIN,PROT
count,13601.000000,13601.000000,13601.000000,13601.000000,13601.000000,3460.000000,7046.000000,1.360100e+04,13601.000000,10141.000000,13601.000000,0.0,13601.000000
mean,2784.055217,1.048379,2019.163664,8.189104,16.749136,0.601156,0.561674,1.383450e+06,8.125138,4.395227,7.293949,NaN,0.910007
std,1895.679710,0.214573,2.424108,1.636388,7.689068,0.532710,0.394032,1.001342e+06,4.390623,1.533837,11.075818,NaN,0.914179
min,1004.000000,1.000000,2016.000000,1.000000,1.000000,0.000000,0.150000,1.846000e+04,3.000000,0.000000,0.000000,NaN,0.000000
25%,1702.000000,1.000000,2016.000000,8.000000,11.000000,0.000000,0.270000,5.714460e+05,4.000000,4.000000,0.000000,NaN,0.000000
50%,2308.000000,1.000000,2018.000000,8.000000,17.000000,1.000000,0.436667,1.345270e+06,8.000000,4.000000,0.000000,NaN,1.000000
75%,3114.000000,1.000000,2022.000000,9.000000,23.000000,1.000000,0.720000,1.735944e+06,13.000000,6.000000,24.000000,NaN,2.000000
max,9999.000000,2.000000,2022.000000,12.000000,31.000000,2.000000,2.000000,4.155843e+06,17.000000,7.000000,28.000000,NaN,2.000000


In [6]:
df_FL_Benthic_Cover.columns

Index(['time', 'latitude', 'longitude', 'REGION', 'PRIMARY_SAMPLE_UNIT',
       'STATION_NR', 'YEAR', 'MONTH', 'DAY', 'Date_UTC', 'HABITAT_CD',
       'HABITAT_TYPE', 'STRAT', 'Description', 'RUGOSITY_CD', 'WTD_RUG',
       'MAPGRID_NR', 'SUB_REGION_NAME', 'SUB_REGION_NR', 'ZONE_NAME',
       'ZONE_NR', 'MPA_NAME', 'MPA_NR', 'ADMIN', 'PROT', 'DEPTH_STRAT',
       'MIN_DEPTH', 'MAX_DEPTH', 'METERS_COMPLETED', 'COVER_CAT_CD',
       'COVER_CAT_NAME', 'HARDBOTTOM_P', 'SOFTBOTTOM_P', 'RUBBLE_P',
       'accession_url'],
      dtype='object')

The `CRCP_Benthic_Cover_Florida_7018_0ee5_9488.csv` dataset contains 13,602 benthic survey records from Florida coral reefs collected between 2016 and 2022. Each row represents an observation at a specific station and date, with geographic coordinates (`latitude`, `longitude`), regional and zoning information (`REGION`, `SUB_REGION_NAME`, `ZONE_NAME`, `MPA_NAME`), and details about the survey such as depth, transect length, and reef complexity (`RUGOSITY_CD`, `WTD_RUG`). Benthic cover is recorded by category (`COVER_CAT_NAME`) along with percent cover of hard bottom, soft bottom, and rubble. Several columns contain missing values or are stored as text rather than numeric values, and the first row of the file contains unit labels rather than actual data. The dataset is suitable for spatial and temporal analysis of coral reef benthic composition, habitat type distribution, and substrate characteristics across Florida reef sites.

**NOTEs**: 
* The first row is a units row (does not contain data)
> `df = pd.read_csv("CRCP_Benthic_Cover_Florida_7018_0ee5_9488.csv", skiprows=1)`

* Some missing data: Columns like `RUGOSITY_CD`, `DEPTH_STRAT`, `Description`, and `ZONE_NAME` have lots of missing values


### Puerto Rico Data

In [7]:
df_PR_Benthic_Cover.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10656 entries, 0 to 10655
Data columns (total 34 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   time                 10656 non-null  object 
 1   latitude             10656 non-null  object 
 2   longitude            10656 non-null  object 
 3   REGION               10655 non-null  object 
 4   PRIMARY_SAMPLE_UNIT  10655 non-null  float64
 5   STATION_NR           10655 non-null  float64
 6   YEAR                 10655 non-null  float64
 7   MONTH                10655 non-null  float64
 8   DAY                  10655 non-null  float64
 9   Date_UTC             10656 non-null  object 
 10  HABITAT_CD           10655 non-null  object 
 11  STRAT                10655 non-null  object 
 12  RUGOSITY_CD          0 non-null      float64
 13  WTD_RUG              4754 non-null   float64
 14  MEAN_RUG             5862 non-null   object 
 15  MAPGRID_NR           10628 non-null 

In [8]:
df_PR_Benthic_Cover.head()

,time,latitude,longitude,REGION,PRIMARY_SAMPLE_UNIT,STATION_NR,YEAR,MONTH,DAY,Date_UTC,...,DEPTH_STRAT,MIN_DEPTH,MAX_DEPTH,METERS_COMPLETED,COVER_CAT_CD,COVER_CAT_NAME,HARDBOTTOM_P,SOFTBOTTOM_P,RUBBLE_P,accession_url
0,UTC,degrees_north,degrees_east,NaN,NaN,NaN,NaN,NaN,NaN,UTC,...,NaN,m,m,m,NaN,NaN,percent,percent,percent,NaN
1,2019-07-18T00:00:00Z,18.14334361,-67.30063227,PRICO,6243.0,1.0,2019.0,7.0,18.0,2019-07-18T00:00:00Z,...,SHLW,8.839199717,9.448799698,15,BAR SUB.,Bare Substrate,29,0,0,https://accession.nodc.noaa.gov/0217139
2,2019-07-18T00:00:00Z,18.14334361,-67.30063227,PRICO,6243.0,1.0,2019.0,7.0,18.0,2019-07-18T00:00:00Z,...,SHLW,8.839199717,9.448799698,15,CYA SPE.,Cyanophyta spp,4,0,0,https://accession.nodc.noaa.gov/0217139
3,2019-07-18T00:00:00Z,18.14334361,-67.30063227,PRICO,6243.0,1.0,2019.0,7.0,18.0,2019-07-18T00:00:00Z,...,SHLW,8.839199717,9.448799698,15,DIC SPE.,Dictyota spp,18,0,0,https://accession.nodc.noaa.gov/0217139
4,2019-07-18T00:00:00Z,18.14334361,-67.30063227,PRICO,6243.0,1.0,2019.0,7.0,18.0,2019-07-18T00:00:00Z,...,SHLW,8.839199717,9.448799698,15,GOR GORG,Gorgonians,1,0,0,https://accession.nodc.noaa.gov/0217139


In [9]:
df_PR_Benthic_Cover.describe()

,PRIMARY_SAMPLE_UNIT,STATION_NR,YEAR,MONTH,DAY,RUGOSITY_CD,WTD_RUG,MAPGRID_NR,SUB_REGION_NR,ZONE_NAME,ZONE_NR,MPA_NAME,MPA_NR,PROT
count,10655.000000,10655.0,10655.000000,10655.000000,10655.000000,0.0,4754.000000,1.062800e+04,0.0,0.0,0.0,0.0,0.0,0.0
mean,6273.545378,1.0,2018.363210,8.110371,17.405913,NaN,0.444762,8.270768e+06,NaN,NaN,NaN,NaN,NaN,NaN
std,2407.341171,0.0,3.435621,1.806798,8.820304,NaN,0.279351,2.906061e+06,NaN,NaN,NaN,NaN,NaN,NaN
min,1000.000000,1.0,2014.000000,4.000000,1.000000,NaN,0.100000,3.642914e+06,NaN,NaN,NaN,NaN,NaN,NaN
25%,6062.000000,1.0,2014.000000,7.000000,11.000000,NaN,0.235417,5.321685e+06,NaN,NaN,NaN,NaN,NaN,NaN
50%,6321.000000,1.0,2019.000000,8.000000,18.000000,NaN,0.354167,7.981655e+06,NaN,NaN,NaN,NaN,NaN,NaN
75%,9141.000000,1.0,2021.000000,9.000000,25.000000,NaN,0.608333,1.098570e+07,NaN,NaN,NaN,NaN,NaN,NaN
max,9359.000000,1.0,2023.000000,12.000000,31.000000,NaN,1.566667,1.358056e+07,NaN,NaN,NaN,NaN,NaN,NaN


In [10]:
df_PR_Benthic_Cover.columns

Index(['time', 'latitude', 'longitude', 'REGION', 'PRIMARY_SAMPLE_UNIT',
       'STATION_NR', 'YEAR', 'MONTH', 'DAY', 'Date_UTC', 'HABITAT_CD', 'STRAT',
       'RUGOSITY_CD', 'WTD_RUG', 'MEAN_RUG', 'MAPGRID_NR', 'SUB_REGION_NAME',
       'SUB_REGION_NR', 'ZONE_NAME', 'ZONE_NR', 'MPA_NAME', 'MPA_NR', 'ADMIN',
       'PROT', 'DEPTH_STRAT', 'MIN_DEPTH', 'MAX_DEPTH', 'METERS_COMPLETED',
       'COVER_CAT_CD', 'COVER_CAT_NAME', 'HARDBOTTOM_P', 'SOFTBOTTOM_P',
       'RUBBLE_P', 'accession_url'],
      dtype='object')

The `CRCP_Benthic_Cover_Puerto_Rico_07c6_3c17_78b6`.csv dataset contains 10,656 benthic survey records from Puerto Rico coral reef sites, collected between 2014 and 2023. Each entry records observations at a specific station and date, including geographic coordinates (`latitude`, `longitude`), regional information (`REGION`, `SUB_REGION_NAME`), and survey details such as depth, transect length, and limited reef complexity metrics (`WTD_RUG`, `MEAN_RUG`). Percent cover of different benthic categories is reported via `COVER_CAT_NAME` along with substrate percentages for hard bottom, soft bottom, and rubble. Many columns related to zoning, MPAs, and rugosity are empty or sparsely filled, and the first row contains unit labels rather than actual data. Like the Florida dataset, this collection is suitable for analyzing spatial and temporal patterns in benthic composition, substrate characteristics, and habitat types across Puerto Rico reef sites.

*** NOTES ***:
* The first row contains units/label
* Many numeric-looking columns are stored as object strings, e.g., latitude, longitude, MIN_DEPTH, MAX_DEPTH, HARDBOTTOM_P, SOFTBOTTOM_P, RUBBLE_P.
> These need conversion to float for calculations or plots.

* Some missing data: many columns like `RUGOSITY_CD`, `SUB_REGION_NR`, `ZONE_NAME`, `MPA_NAME`, `PROT` are empty

### USVI Data

In [11]:
df_USVI_Benthic_Cover.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26326 entries, 0 to 26325
Data columns (total 40 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   time                         26326 non-null  object 
 1   latitude                     26326 non-null  object 
 2   longitude                    26326 non-null  object 
 3   REGION                       26325 non-null  object 
 4   REGION_DESCRIPTION           26325 non-null  object 
 5   PRIMARY_SAMPLE_UNIT          26325 non-null  float64
 6   STATION_NR                   26325 non-null  float64
 7   YEAR                         26325 non-null  float64
 8   MONTH                        26325 non-null  float64
 9   DAY                          26325 non-null  float64
 10  Date_UTC                     26326 non-null  object 
 11  HABITAT_CD                   26325 non-null  object 
 12  HABITAT_TYPE                 26325 non-null  object 
 13  STRAT           

In [12]:
df_USVI_Benthic_Cover.head()

,time,latitude,longitude,REGION,REGION_DESCRIPTION,PRIMARY_SAMPLE_UNIT,STATION_NR,YEAR,MONTH,DAY,...,DEPTH_STRAT_DESCRIPTION,MIN_DEPTH,MAX_DEPTH,METERS_COMPLETED,COVER_CAT_CD,COVER_CAT_NAME,HARDBOTTOM_P,SOFTBOTTOM_P,RUBBLE_P,accession_url
0,UTC,degrees_north,degrees_east,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,m,m,m,NaN,NaN,percent,percent,percent,NaN
1,2013-07-11T00:00:00Z,18.35835,-64.84812,STTSTJ,St. Thomas and St. John - U.S. Virgin Islands,8781.0,1.0,2013.0,7.0,11.0,...,shallow (0-11.9 m),3.047999902,5.791199815,20,DIC SPE.,Dictyota spp,2,0,0,https://accession.nodc.noaa.gov/0224205
2,2013-07-11T00:00:00Z,18.35835,-64.84812,STTSTJ,St. Thomas and St. John - U.S. Virgin Islands,8781.0,1.0,2013.0,7.0,11.0,...,shallow (0-11.9 m),3.047999902,5.791199815,20,MIL SPE.,Millepora spp,1,0,0,https://accession.nodc.noaa.gov/0224205
3,2013-07-11T00:00:00Z,18.35835,-64.84812,STTSTJ,St. Thomas and St. John - U.S. Virgin Islands,8781.0,1.0,2013.0,7.0,11.0,...,shallow (0-11.9 m),3.047999902,5.791199815,20,OTH SPE.,Other species,2,0,0,https://accession.nodc.noaa.gov/0224205
4,2013-07-11T00:00:00Z,18.35835,-64.84812,STTSTJ,St. Thomas and St. John - U.S. Virgin Islands,8781.0,1.0,2013.0,7.0,11.0,...,shallow (0-11.9 m),3.047999902,5.791199815,20,POR ASTR,Porites astreoides,2,0,0,https://accession.nodc.noaa.gov/0224205


In [13]:
df_USVI_Benthic_Cover.describe()

,PRIMARY_SAMPLE_UNIT,STATION_NR,YEAR,MONTH,DAY,RUGOSITY_CD,WTD_RUG,MAPGRID_NR,SUB_REGION_NR,ZONE_NAME,ZONE_NR,MPA_NAME,MPA_NR,PROT
count,26325.000000,26325.0,26325.000000,26325.000000,26325.000000,0.0,12847.000000,2.630800e+04,0.0,0.0,0.0,0.0,0.0,4722.0
mean,6071.719962,1.0,2018.214207,6.877379,13.472365,NaN,0.491789,5.437122e+05,NaN,NaN,NaN,NaN,NaN,0.0
std,2492.412537,0.0,3.325142,0.825953,7.175845,NaN,0.325364,2.535643e+05,NaN,NaN,NaN,NaN,NaN,0.0
min,1000.000000,1.0,2013.000000,6.000000,1.000000,NaN,0.100000,1.988700e+04,NaN,NaN,NaN,NaN,NaN,0.0
25%,4349.000000,1.0,2015.000000,6.000000,8.000000,NaN,0.270000,3.649205e+05,NaN,NaN,NaN,NaN,NaN,0.0
50%,5318.000000,1.0,2019.000000,7.000000,13.000000,NaN,0.416667,4.465880e+05,NaN,NaN,NaN,NaN,NaN,0.0
75%,8928.000000,1.0,2021.000000,8.000000,17.000000,NaN,0.639583,7.914050e+05,NaN,NaN,NaN,NaN,NaN,0.0
max,9853.000000,1.0,2023.000000,8.000000,31.000000,NaN,3.133333,1.040966e+06,NaN,NaN,NaN,NaN,NaN,0.0


In [14]:
df_USVI_Benthic_Cover.columns

Index(['time', 'latitude', 'longitude', 'REGION', 'REGION_DESCRIPTION',
       'PRIMARY_SAMPLE_UNIT', 'STATION_NR', 'YEAR', 'MONTH', 'DAY', 'Date_UTC',
       'HABITAT_CD', 'HABITAT_TYPE', 'STRAT', 'STRAT_Description',
       'RUGOSITY_CD', 'WTD_RUG', 'MEAN_RUG', 'MAPGRID_NR', 'SUB_REGION_NAME',
       'SUB_REGION_NAME_DESCRIPTION', 'SUB_REGION_NR', 'ZONE_NAME', 'ZONE_NR',
       'MPA_NAME', 'MPA_NR', 'ADMIN', 'administration_description', 'PROT',
       'DEPTH_STRAT', 'DEPTH_STRAT_DESCRIPTION', 'MIN_DEPTH', 'MAX_DEPTH',
       'METERS_COMPLETED', 'COVER_CAT_CD', 'COVER_CAT_NAME', 'HARDBOTTOM_P',
       'SOFTBOTTOM_P', 'RUBBLE_P', 'accession_url'],
      dtype='object')

The `CRCP_Benthic_Cover_USVI_6400_ac60_8c98`.csv dataset contains 26,326 benthic survey records from U.S. Virgin Islands reef sites collected between 2013 and 2023. Each record includes geographic coordinates (`latitude`, `longitude`), regional and habitat classifications (`REGION_DESCRIPTION`, `HABITAT_TYPE`, `STRAT_Description`), and survey details such as depth range, transect length, and reef complexity metrics (`WTD_RUG`, `MEAN_RUG`). Percent cover for different benthic categories is provided through `COVER_CAT_NAME` along with substrate percentages for hard bottom, soft bottom, and rubble. Several zoning and MPA-related fields are entirely empty, and `RUGOSITY_CD` contains no values, though weighted rugosity is partially available.

*** Notes ***:
* The first row contains unit labels rather than actual data
* Multiple numeric columns are stored as text

## Overall Data Summary
---

### What the datasets contain
Across the three regions, the datasets provide:
* ~50,000+ benthic survey records total
* Time span:
  * Florida: 2016–2022
  * Puerto Rico: 2014–2023
  * USVI: 2013–2023

* Spatial coverage:
  * Florida Keys
  * Puerto Rico reef systems
  * U.S. Virgin Islands reef systems

* Each row represents:
  * A specific station
  * On a specific date
  * With a specific benthic cover category
  * And associated percent substrate cover


### Key Things for Our Question
1. Percent Live Coral Cover
   * `COVER_CAT_NAME` identifies species or benthic groups
   * `HARDBOTTOM_P`, `SOFTBOTTOM_P`, and `RUBBLE_P` contain percent cover values
    > By filtering for live coral species and summing their percent cover by station and year, we can calculate total live coral cover and track changes over time

2. Tempurature Information
   * `YEAR`, `MONTH`, and `Date_UTC` allow us to build time series of coral cover
    > These variables enable analysis of decline and recovery patterns following stress events
3. Algal Cover
   * Algal groups are also listed under COVER_CAT_NAME (e.g., macroalgae, turf algae, cyanobacteria)
  > Summing these categories allows comparison of coral and algal trends to detect potential phase shifts
4. Reef Complexity
   * WTD_RUG and MEAN_RUG measure structural complexity
  > These variables can help assess whether more structurally complex reefs recover differently than flatter reefs
5. Thermal Stress
   * We may need to find some more data for this one. 